tensorflow/core/kernels/mkl_conv_ops.cc

```diff
@@ -488,17 +489,22 @@ class MklConvFwdPrimitiveFactory : public MklPrimitiveFactory<float> {
 
     // Generate keys for post-ops
     for (auto const& post_op_param : convFwdDims.post_op_params) {
+      key_creator.AddAsKey(post_op_param.name);
       if (post_op_param.name == "activation") {
         DCHECK_EQ(post_op_param.param.size(), 3);
+        for (auto& param : post_op_param.param) {
```

Comment:
Can this loop be pulled up to before the `if`/`else` (i.e., before line 493)? `AddAsKey` wouldn't be called if `post_op_param.param` is empty anyway.

Reply:
NO. If `post_op_param.name == "output_scale"`, `post_op_param.param` is not empty and we shouldn't call `AddAsKey`.
Actually, `post_op_param.param` is used to generate op_scale and partial key. When it is `output_scale`, we use `post_op_param.param` only for op_scale and use `post_op_param.partial_key` as a partial key.

Reply:
Got it. Thank you for the explanation!